homeassistant/components/mqtt/models.py

```diff
@@ -313,8 +358,11 @@ def process_write_state_requests(self, msg: MQTTMessage) -> None:
                 )
 
     @callback
-    def write_state_request(self, entity: Entity) -> None:
+    def write_state_request(self, entity: MqttEntity) -> None:
         """Register write state request."""
+        if not entity.monitor.assume_has_changed:
+            # no change detected skip state write request
+            return
```

Comment:
We only mark an entity for a state write request if it is assumed that it has changed.